remote-api/src/main/java/org/alfresco/repo/webdav/LockInfoImpl.java

```diff
@@ -390,8 +390,14 @@ public long getRemainingTimeoutSeconds()
         else
         {
             Date now = dateNow();
-            long timeout = ((expires.getTime() - now.getTime()) / 1000);
-            return timeout;
+
+            double remainingTimeoutInMs = expires.getTime() - now.getTime();
+            double remainingTimeoutInS = remainingTimeoutInMs / 1000;
+
+            long timeoutRoundedUp = (long) Math.ceil(remainingTimeoutInS);
+            long minTimeout = 0L;
+
+            return Math.max(minTimeout, timeoutRoundedUp);
         }
```

Comment:
I've tested the formula, it is working fine. I've renamed the variable so that next person reading it won't have to analyse the formula itself/to make analysis easier.

Reply:
I've tested the formula, it is working fine. I've renamed the variable so that next person reading it won't have to analyse the formula itself/to make analysis easier.